### ISSUE TITLE
PHP Warnings about undefined array keys

### ISSUE THREAD

**Issue by fiwswe** [dokufreaks/plugin-pagelist#131]
DokuWiki 2022-07-31 "Igor", PHP 8.0.22 (PHP-FPM, OpenBSD 7.1, httpd)

I am getting these PHP Warning on a page using the PageList Plugin:
```
PHP message: PHP Warning:  Undefined array key "exists" in /dokuwiki-test/lib/plugins/pagelist/helper.php on line 441
PHP message: PHP Warning:  Undefined array key "class" in /dokuwiki-test/lib/plugins/pagelist/helper.php on line 253
PHP message: PHP Warning:  Undefined array key "titleimage" in /dokuwiki-test/lib/plugins/pagelist/helper.php on line 319
PHP message: PHP Warning:  Undefined array key "section" in /dokuwiki-test/lib/plugins/pagelist/helper.php on line 336
```

In general I'd think a pattern such as `if (array_key_exists('key', $array) && $array['key'])` or `if (isset($array['key']))` could be used but I don't understand the code enough to make concrete suggestions. What _should_ happen when no element with that key exists?

These keys seem to be set by the plugin but apparently not always.

Initializing the array with empty (`null`) elements for all keys might also be an option. Again I don't understand enough to be able to judge the semantic ramifications.

fiwswe

**Comment by Klap-in** [comment]
I did make an update for this plugin. I have not checked now if I already catch these issues.
Eventually you can try: https://github.com/dokufreaks/plugin-pagelist/archive/refs/heads/cleanup.zip

**Comment by fiwswe** [comment]
Ok, I replaced the plugin with your new version and tested again (using `&purge=true` to avoid caching issues):

No more PHP warnings! So your changes fixed these problems, thanks.

Are you planing to officially release this cleaned up version?

Thx.
fiwswe

**Comment by tqre-mc** [comment]
Hi, stumbled on the same issue, and monkeypatched the plugin, and now the PHP errors are gone. Official release would indeed be nice to be able to update from plugin manager.

**Comment by Klap-in** [comment]
Fixed by #130
